FEXCore/Source/Utils/FutexSpinWait.h

```diff
@@ -0,0 +1,282 @@
+#include <atomic>
+#include <chrono>
+#include <mutex>
+#include <type_traits>
+
+namespace FEXCore::Utils {
+#define SPINLOOP_8BIT " \
+  /* Prime the exclusive monitor with the passed in address. */ \
+  ldaxrb %w[Tmp], [%[Futex]]; \
+  /* WFE will wait for either the memory to change or spurious wake-up. */ \
+  wfe; \
+  /* Load with acquire to get the result of memory. */ \
+  ldarb %w[Result], [%[Futex]]; \
+"
+#define SPINLOOP_16BIT " \
+  /* Prime the exclusive monitor with the passed in address. */ \
+  ldaxrh %w[Tmp], [%[Futex]]; \
+  /* WFE will wait for either the memory to change or spurious wake-up. */ \
+  wfe; \
+  /* Load with acquire to get the result of memory. */ \
+  ldarh %w[Result], [%[Futex]]; \
+"
+#define SPINLOOP_32BIT " \
+  /* Prime the exclusive monitor with the passed in address. */ \
+  ldaxr %w[Tmp], [%[Futex]]; \
+  /* WFE will wait for either the memory to change or spurious wake-up. */ \
+  wfe; \
+  /* Load with acquire to get the result of memory. */ \
+  ldar %w[Result], [%[Futex]]; \
+"
+#define SPINLOOP_64BIT " \
+  /* Prime the exclusive monitor with the passed in address. */ \
+  ldaxr %x[Tmp], [%[Futex]]; \
+  /* WFE will wait for either the memory to change or spurious wake-up. */ \
+  wfe; \
+  /* Load with acquire to get the result of memory. */ \
+  ldar %x[Result], [%[Futex]]; \
+"
+
+  /**
+   * @brief This provides routines to implement implement an "efficient spin-loop" using ARM's WFE and exclusive monitor interfaces.
+   *
+   * Spin-loops on mobile devices with a battery can be a bad idea as they burn a bunch of power. This attempts to mitigate some of the impact
+   * by putting the CPU in to a lower-power state using WFE.
+   * On platforms tested, WFE will put the CPU in to a lower power state for upwards of 52ns per WFE. Which isn't a significant amount of time
+   * but should still have power savings. Ideally WFE would be able to keep the CPU in a lower power state for longer. This also has the added benefit
+   * that atomics aren't abusing the caches when spinning on a cacheline, which has knock-on powersaving benefits.
+   *
+   * FEAT_WFxT adds a new instruction with a timeout, but since the spurious wake-up is so aggressive it isn't worth using.
+   *
+   * It should be noted that this implementation has a few dozen cycles of start-up time. Which means the overhead for invoking this implementation is
+   * slightly higher than a true spin-loop. The hot loop body itself is only three instructions so it is quite efficient.
+   *
+   * On non-ARM platforms it is truly a spin-loop. Which is okay for debugging only.
```

Comment:
Relative clauses are connected using comma:

```suggestion
   * On non-ARM platforms it is truly a spin-loop, which is okay for debugging only.
```